researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/TaskHelper.java

```diff
@@ -195,16 +198,25 @@ void uploadTaskResult(TaskResult taskResult, Archive.WithBridgeConfig withBridge
         String archiveFilename = taskId + "_" + UUID.randomUUID().toString() + ".zip";
 
         bridgeManagerProvider.getUploadManager()
-                .upload(archiveFilename, bridgeArchiveBuilder.build()).toCompletable().await();
-
-        // At this point, the upload request has been processed and saved,
-        // so it is safe to delete the temporary data logger files
-        for (Result result : results) {
-            if (result instanceof FileResult) {
-                FileResult fileResult = (FileResult) result;
-                DataLoggerManager.getInstance().deleteFileStatus(fileResult.getFile());
-            }
-        }
+                .queueUpload(archiveFilename, bridgeArchiveBuilder.build())
+                .doOnSuccess(uploadFile -> {
+                    logger.debug("Attempting upload in io() thread");
+                    bridgeManagerProvider.getUploadManager()
```

Comment:
If upload needs to happen on the IO thread, I think the upload manager should put it on the I/O thread, since it shouldn't be the responsibility of the caller to know to call process upload file on a specific thread.